superset/utils/core.py

```diff
@@ -581,47 +588,60 @@ def base_json_conv(  # pylint: disable=inconsistent-return-statements
         except Exception:  # pylint: disable=broad-except
             return "[bytes]"
 
+    raise TypeError(f"Unserializable object {obj} of type {type(obj)}")
```

Comment:
Previously this would return `None` however there was no way to distinguish between the conversion of `np.array(None)` which returns `None` and an unserializable object.